airflow/traces/otel_tracer.py

```diff
@@ -0,0 +1,316 @@
+        """Produce a span from dag run."""
+        # check if dagrun has configs
+        conf = dagrun.conf
+        trace_id = int(gen_trace_id(dag_run=dagrun, as_int=True))
+        span_id = int(gen_dag_span_id(dag_run=dagrun, as_int=True))
+
+        if conf is not None:
+            traceparent = conf.get(TRACEPARENT)
+            tracestate = conf.get(TRACESTATE)
+
+        tracer = self.get_tracer(component=component, span_id=span_id, trace_id=trace_id)
+
+        if self.tag_string and tracestate:
+            tag_string = self.tag_string + "," + tracestate
+        else:
+            tag_string = self.tag_string or tracestate
+
+        if span_name is None:
+            span_name = dagrun.dag_id
+
+        if links is not None:
+            _links = gen_links_from_kv_list(links)
+        else:
+            _links = []
+
+        _links.append(
+            Link(
+                context=trace.get_current_span().get_span_context(),
+                attributes={"meta.annotation_type": "link", "from": "parenttrace"},
+            )
+        )
+
+        if traceparent is not None:
+            # add the trace parent as the link
+            _links.append(gen_link_from_traceparent(traceparent))
+
+        span_ctx = SpanContext(
+            trace_id=INVALID_TRACE_ID, span_id=INVALID_SPAN_ID, is_remote=True, trace_flags=TraceFlags(0x01)
+        )
+        ctx = trace.set_span_in_context(NonRecordingSpan(span_ctx))
+        span = tracer.start_as_current_span(
+            name=span_name,
+            context=ctx,
+            links=_links,
+            start_time=datetime_to_nano(dagrun.queued_at),
+            attributes=parse_tracestate(tag_string),
+        )
+        return span
+
+    def start_span_from_taskinstance(
+        self,
+        ti,
+        span_name: str | None = None,
+        component: str = "taskinstance",
+        child: bool = False,
+        links=None,
+    ):
+        """
+        Create and start span from given task instance.
+
+        Essentially the span represents the ti itself if child == True, it will create a 'child' span under the given span.
+        """
+        dagrun = ti.dag_run
+        trace_id = int(gen_trace_id(dag_run=dagrun, as_int=True))
+        span_id = int(gen_span_id(ti=ti, as_int=True))
+        if span_name is None:
+            span_name = ti.task_id
+
+        if child is False:
+            parent_id = int(gen_dag_span_id(dag_run=dagrun, as_int=True))
+        else:
+            parent_id = span_id
+
+        span_ctx = SpanContext(
+            trace_id=trace_id, span_id=parent_id, is_remote=True, trace_flags=TraceFlags(0x01)
+        )
+
+        if links is not None:
+            _links = gen_links_from_kv_list(links)
+        else:
+            _links = []
+
+        a_link = Link(
+            context=trace.get_current_span().get_span_context(),
+            attributes={"meta.annotation_type": "link", "from": "parenttrace"},
+        )
+        _links.append(a_link)
```

Comment:
I think maybe something happened here, I'm pretty sure you already changed this too....
```suggestion
        _links.append(
            Link(
                context=trace.get_current_span().get_span_context(),
                attributes={"meta.annotation_type": "link", "from": "parenttrace"},
             )
         )
```